src/iam/iam-base.ts

```diff
@@ -70,8 +70,24 @@ export type Transaction = {
   from: string;
 };
 
+export interface AssetProfile {
+  name?: string;
+  icon?: string;
+}
+
+export interface AssetProfiles {
+  [key:string]: AssetProfile;
+}
+
+export interface Profile {
+  name?: string;
+  birthdate?: string;
+  address?: string;
+  assetProfiles?: AssetProfiles;
+}
+
```

Comment:
Looks good. Perhaps consider moving these interfaces (and maybe the `ClaimData` interface to a separate file), up to you.